spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/SimpleDirectMessageListenerContainer.java

```diff
@@ -0,0 +1,218 @@
+ * @author Gary Russell
+ * @since 1.6
+ *
+ */
+public class SimpleDirectMessageListenerContainer extends AbstractMessageListenerContainer {
+
+	private final List<SimpleConsumer> consumers = new ArrayList<SimpleConsumer>();
+
+	private final MessagePropertiesConverter messagePropertiesConverter = new DefaultMessagePropertiesConverter();
+
+	private volatile TaskExecutor taskExecutor;
+
+	private volatile int consumersPerQueue = 1;
+
+	private volatile int prefetch = 1;
+
+	private volatile boolean requeueRejected = false;
+
+	/**
+	 * Create an instance with the provided connection factory.
+	 * @param connectionFactory the connection factory.
+	 */
+	public SimpleDirectMessageListenerContainer(ConnectionFactory connectionFactory) {
+		setConnectionFactory(connectionFactory);
+	}
+
+	@Override
+	public final void setConnectionFactory(ConnectionFactory connectionFactory) {
+		super.setConnectionFactory(connectionFactory);
+	}
+
+	/**
+	 * Set a task executor for the container - used to create the consumers not at
+	 * runtime.
+	 * @param taskExecutor the task excutor.
+	 */
+	public void setTaskExecutor(TaskExecutor taskExecutor) {
+		this.taskExecutor = taskExecutor;
+	}
+
+	/**
+	 * Each queue runs in its own consumer; set this property to create multiple
+	 * consumers for each queue.
+	 * @param consumersPerQueue the consumers per queue.
+	 */
+	public void setConsumersPerQueue(int consumersPerQueue) {
+		this.consumersPerQueue = consumersPerQueue;
+	}
+
+	/**
+	 * Set to false to requeue failed deliveries.
+	 * @param requeueRejected the requeue rejected.
+	 */
+	public void setRequeueRejected(boolean requeueRejected) {
+		this.requeueRejected = requeueRejected;
+	}
+
+	@Override
+	protected void doInitialize() throws Exception {
+		Assert.notNull(getQueueNames(), "queue(s) are required");
+	}
+
+	@Override
+	protected void doStart() throws Exception {
+		super.doStart();
+		if (this.taskExecutor == null) {
+			this.taskExecutor = new SimpleAsyncTaskExecutor((getBeanName() == null ? "container" : getBeanName()) + "-");
+		}
+		this.taskExecutor.execute(new Runnable() {
+
+			boolean initialized = false;
+
+			@Override
+			public void run() {
+				while (!initialized) {
+					try {
+						for (int i = 0; i < consumersPerQueue; i++) {
+							for (String queue : getQueueNames()) {
+								Connection connection = getConnectionFactory().createConnection();
+								Channel channel = connection.createChannel(false);
+								channel.basicQos(prefetch);
+								RabbitUtils.setPhysicalCloseRequired(true);
+								SimpleConsumer consumer = new SimpleConsumer(channel, queue);
+								consumers.add(consumer);
+								channel.basicConsume(queue, consumer);
+							}
+						}
+					}
+					catch (Exception e) {
+						logger.error("Error creating consumer", e);
+						doShutdown();
+						try {
+							Thread.sleep(5000);
+						}
+						catch (InterruptedException e1) {
+							Thread.currentThread().interrupt();
+						}
+						continue;
```

Comment:
What is the point of this `continue`? Retry?
The simplified version of `restart()`? :smile: